models/models.reviews.js

```diff
@@ -15,18 +15,32 @@ exports.fetchReviewById = (review_id) => {
     });
 };
 
-exports.fetchReviews = () => {
-  return db
-    .query(
-      `SELECT reviews.*, COUNT(comments.review_id) AS comment_count FROM reviews 
-      LEFT JOIN comments 
-      ON reviews.review_id = comments.review_id 
-      GROUP BY reviews.review_id 
-      ORDER BY reviews.created_at DESC;`
-    )
-    .then((res) => {
-      return res.rows;
-    });
+exports.fetchReviews = (
+  category = "",
+  sort_by = "created_at",
+  order = "DESC"
+) => {
+  let categoryAdd = "";
+
+  if (category.length) {
+    categoryAdd = `WHERE category LIKE '%${category}%'`;
+  }
+
+  let queryStr = `SELECT reviews.*, COUNT(comments.review_id) AS comment_count FROM reviews 
+  LEFT JOIN comments 
+  ON reviews.review_id = comments.review_id ${categoryAdd}
+  GROUP BY reviews.review_id 
+  ORDER BY reviews.${sort_by} ${order};`;
+
```

Comment:
There is no protection against SQL injection here so you will need some logic to check that these queries are all valid and acceptable. If they are not, then you will need to reject the Promise and handle the custom error